src/Intercom.Tests/Clients/CompanyClientTest.cs

```diff
@@ -23,38 +23,40 @@ public CompanyClientTest()
         }
 
         [Test()]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void Create_WithNull_ThrowException()
         {
-            companyClient.Create(null);
+            Assert.Throws<ArgumentNullException>(() => companyClient.Create(null));
         }
 
         [Test()]
-        [ExpectedException(typeof(ArgumentException))]
         public void Delete_MoreThan100CustomAtt_ThrowException()
         {
             Dictionary<string, object> custom_attributes = new Dictionary<string, object>();
 
             for (int i = 0; i < 105; i++)
                 custom_attributes.Add("field", "value");
 
-            companyClient.Create(new Company() { custom_attributes = custom_attributes } );
+            Assert.Throws<ArgumentException>(() => 
+            {
+                companyClient.Create(new Company() { custom_attributes = custom_attributes });
+            });
         }
 
         [Test()]
-        [ExpectedException(typeof(ArgumentException))]
         public void Create_CustomAttInvalidChars_ThrowException()
         {
             Dictionary<string, object> custom_attributes = new Dictionary<string, object>();
             custom_attributes.Add("invalid.ch$ar", "invalid");
-            companyClient.Create(new Company() { custom_attributes = custom_attributes } );
+            Assert.Throws<ArgumentException>(() =>
+            {
+                companyClient.Create(new Company() { custom_attributes = custom_attributes });
+            });
         }
 
         [Test()]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void Update_WithNull_ThrowException()
         {
-            companyClient.Update(null);
+            Assert.Throws<ArgumentNullException>(() => companyClient.Update(null));
         }
     }
 }
```

Comment:
Do you think we need to add a test for something like plan here? just wondering since the way plan is a string and returns an object, it caused issues before. Maybe now is not the time? but it would be good to try and catch edge cases like that i.e. create company without plan and with plan tests to make sure they work?